### PR TITLE
Implement SPV_NV_bindless_texture related changes

### PR DESCRIPTION
**Pull request by pmistryNV**
@alan-baker kindly let me know where should I add test cases for this extension?
Note: Previous PR https://github.com/KhronosGroup/SPIRV-Tools/pull/4597 was closed. I have addressed all the comments except adding test cases.

### CODE REVIEW

File: source/val/validate_decorations.cpp [hunk 1]
```diff
@@ -190,6 +190,12 @@ uint32_t getBaseAlignment(uint32_t member_id, bool roundUp,
   // Minimal alignment is byte-aligned.
   uint32_t baseAlignment = 1;
   switch (inst->opcode()) {
+    case SpvOpTypeSampledImage:
```
Comment (alan-baker) [inline source/val/validate_decorations.cpp:193]: Tests for this file are in test/val/val_decorations_test.cpp

File: source/val/validate_image.cpp [hunk 2]
```diff
@@ -927,7 +927,8 @@ spv_result_t ValidateTypeSampledImage(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
-bool IsAllowedSampledImageOperand(SpvOp opcode) {
+bool IsAllowedSampledImageOperand(SpvOp opcode,
```
Comment (alan-baker) [inline source/val/validate_image.cpp:930]: Tests should go in test/val/val_image_test.cpp

File: source/val/validate_instruction.cpp [hunk 3]
```diff
@@ -483,6 +483,21 @@ spv_result_t InstructionPass(ValidationState_t& _, const Instruction* inst) {
     if (auto error = LimitCheckNumVars(_, inst->id(), storage_class)) {
       return error;
     }
+  } else if (opcode == SpvOpSamplerImageAddressingModeNV) {
```
Comment (alan-baker) [inline source/val/validate_instruction.cpp:486]: These also belong in test/val/val_image_test.cpp

File: source/val/validate_layout.cpp [hunk 4]
```diff
@@ -363,6 +363,7 @@ spv_result_t ModuleLayoutPass(ValidationState_t& _, const Instruction* inst) {
     case kLayoutExtensions:
     case kLayoutExtInstImport:
     case kLayoutMemoryModel:
+    case kLayoutSamplerImageAddressMode:
```
Comment (alan-baker) [inline source/val/validate_layout.cpp:366]: Tests belong in test/val/val_layout_test.cpp

File: source/val/validate_logicals.cpp [hunk 5]
```diff
@@ -170,6 +170,16 @@ spv_result_t LogicalsPass(ValidationState_t& _, const Instruction* inst) {
             break;
           }
 
+          case SpvOpTypeSampledImage:
```
Comment (alan-baker) [inline source/val/validate_logicals.cpp:173]: Tests should go test/val/val_logicals_test.cpp

File: source/val/validate_type.cpp [hunk 6]
```diff
@@ -425,8 +425,28 @@ spv_result_t ValidateTypeStruct(ValidationState_t& _, const Instruction* inst) {
     _.RegisterStructTypeWithBuiltInMember(struct_id);
   }
 
+  const auto isOpaqueType = [&_](const Instruction *inst) {
```
Comment (alan-baker) [inline source/val/validate_type.cpp:428]: Tests should go with test/val_ids_test.cpp

**Comment by pmistryNV** [comment]
@alan-baker CI builds are failing, but I am not able to access the failure logs. For example for CI-ubuntu-clang-release "http://fusion/runanalysis/info/prod%3Agraphics_shader_compiler%2Fspirv-tools%2Flinux-clang-release%2Fpresubmit/prod%3Agraphics_shader_compiler%2Fspirv-tools%2Flinux-clang-release%2Fpresubmit/KOKORO/8653d6e5-37d9-4c08-9dd8-d1c8d9590c5b/0/prod%3Agraphics_shader_compiler%2Fspirv-tools%2Flinux-clang-release%2Fpresubmit/Targets" this link doesn't work for me. How to I access it?

File: source/val/validate_instruction.cpp [hunk 7]
```diff
@@ -483,6 +483,22 @@ spv_result_t InstructionPass(ValidationState_t& _, const Instruction* inst) {
...
```
Comment (alan-baker) [inline source/val/validate_instruction.cpp:486]: Should this also check whether another SpvOpSamplerImageAddressingModeNV instruction has been seen? The extension isn't explicit, but I assume only a single instance of this instruction can be specified.

**Review by alan-baker — COMMENTED** [review]
Almost there, is there a limit on the number of instances of the addressing mode that can be specified? Right now the validator seems to assume the last one will be used.

File: source/val/validate_instruction.cpp (hunk 7 above)
Comment (pmistryNV) [reply source/val/validate_instruction.cpp:486]: yes only one is allowed. I can add a check

File: source/val/validate_instruction.cpp [hunk 8]
```diff
@@ -483,6 +483,22 @@ spv_result_t InstructionPass(ValidationState_t& _, const Instruction* inst) {
...
+    if (!_.HasCapability(SpvCapabilityBindlessTextureNV)) {
+      return _.diag(SPV_ERROR_MISSING_EXTENSION, inst)
+             << "OpSamplerImageAddressingModeNV supported only with extension "
+                "SPV_NV_bindless_texture";
+    }
+    uint32_t bitwidth = inst->GetOperandAs<uint32_t>(0);
+    if (_.samplerimage_variable_address_mode() != 0) {
```
Comment (pmistryNV) [inline source/val/validate_instruction.cpp:493]: @alan-baker this is where it checks to ensure OpSamplerImageAddressingModeNV is called only once. Is this enough?

Comment (alan-baker) [reply source/val/validate_instruction.cpp:493]: I think so, can you just cover it with a test?

Comment (pmistryNV) [reply source/val/validate_instruction.cpp:493]: added a test "NVBindlessAddressModeFromLayoutSpecifiedTwice" in val_layout_test.cpp

**Review by alan-baker — APPROVED** [review]

**Pull request merged**